### PR TITLE
Add assignment and course types

### LINKED ISSUE #2: Add UserCourse, Course, and Assignment Types
### Feature request
------------------------
Add UserCourse, Course, and Assignment Type definitions

### PR DESCRIPTION
**Pull request by NicholasMy**
## Proposed changes
--------------------
Add types for assignments and courses

## Types of changes
--------------------
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist
--------------------
- [x] My changeset covers only what is described above (no extraneous changes)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged downstream

### CODE REVIEW

File: src/types/course.types.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ export type Course = {
   name?: string
   semester?: string
   number?: string
-  startDate?: string
-  endDate?: string
+  startDate?: Date
+  endDate?: Date
```
Comment (jessehartloff) [inline src/types/course.types.ts]: Change these back to string

File: src/types/assignment.types.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+export type Assignment = {
+  id?: number
+  courseId?: number
+  name?: string
+  startDate?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  startDate: string
```

File: src/types/assignment.types.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  dueDate?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  dueDate: string
```

File: src/types/assignment.types.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  endDate?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  endDate: string
```

File: src/types/assignment.types.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  gradingType?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  gradingType: string
```

File: src/types/assignment.types.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  categoryName?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  categoryName: string
```

File: src/types/assignment.types.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  description?: string
+  maxFileSize?: number
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  maxFileSize: number
```

File: src/types/assignment.types.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  maxSubmissions?: number
+  disableHandins?: boolean
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  disableHandins: boolean
```

File: src/types/assignment.types.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  courseId?: number
+  name?: string
+  startDate?: string
+  dueDate?: string
+  endDate?: string
+  gradingType?: string
+  categoryName?: string
+  description?: string
+  maxFileSize?: number
+  maxSubmissions?: number
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  maxSubmissions: number | null
```

File: src/types/assignment.types.ts [hunk 10]
```diff
@@ -0,0 +1,44 @@
+export type Assignment = {
+  id?: number
+  courseId?: number
+  name?: string
+  startDate?: string
+  dueDate?: string
+  endDate?: string
+  gradingType?: string
+  categoryName?: string
+  description?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  description: string | null
```

File: src/types/assignment.types.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
+export type Assignment = {
+  id?: number
+  courseId?: number
+  name?: string
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  name: string
```

File: src/types/assignment.types.ts [hunk 12]
```diff
@@ -0,0 +1,44 @@
+export type Assignment = {
+  id?: number
+  courseId?: number
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  courseId: number
```

File: src/types/assignment.types.ts [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  createdAt?: string
+  updatedAt?: string
+}
+
+
+export type UserCourse = {
+  id?: number
+  userId: number
+  courseId: number
+  dropped?: boolean
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  dropped: boolean
```

File: src/types/assignment.types.ts [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+  level?: GradingType
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  level: GradingType
```

File: src/types/assignment.types.ts [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+  description?: string
+  maxFileSize?: number
+  maxSubmissions?: number
+  disableHandins?: boolean
+  createdAt?: string
+  updatedAt?: string
+}
+
+
+export type UserCourse = {
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]: What is this? Should this be here in this PR?

File: src/types/course.types.ts [hunk 16]
```diff
@@ -0,0 +1,10 @@
+export type Course = {
+  id?: number
+  name?: string
```
Comment (MikeDrewitt) [inline src/types/course.types.ts]:
```suggestion
  name: string
```

File: src/types/course.types.ts [hunk 17]
```diff
@@ -0,0 +1,10 @@
...
+  semester?: string
```
Comment (MikeDrewitt) [inline src/types/course.types.ts]:
```suggestion
  semester: string
```

File: src/types/course.types.ts [hunk 18]
```diff
@@ -0,0 +1,10 @@
...
+  number?: string
```
Comment (MikeDrewitt) [inline src/types/course.types.ts]:
```suggestion
  number: string
```

File: src/types/course.types.ts [hunk 19]
```diff
@@ -0,0 +1,10 @@
...
+  startDate?: string
```
Comment (MikeDrewitt) [inline src/types/course.types.ts]:
```suggestion
  startDate: string
```

File: src/types/course.types.ts [hunk 20]
```diff
@@ -0,0 +1,10 @@
...
+  endDate?: string
```
Comment (MikeDrewitt) [inline src/types/course.types.ts]:
```suggestion
  endDate: string
```

**Review by MikeDrewitt — CHANGES_REQUESTED** [review]

File: src/types/assignment.types.ts [hunk 21]
```diff
@@ -0,0 +1,30 @@
+export type Assignment = {
+  id?: number
+  courseId?: number
+  name?: string
+  startDate?: string
+  dueDate?: string
+  endDate?: string
+  gradingType?: GradingType
```
Comment (MikeDrewitt) [inline src/types/assignment.types.ts]:
```suggestion
  gradingType: GradingType
```

**Review by MikeDrewitt — CHANGES_REQUESTED** [review]

**Review by MikeDrewitt — APPROVED** [review]

**Pull request merged**